clang/test/CodeGen/convergent.cpp

```diff
@@ -0,0 +1,99 @@
+// RUN: %clang_cc1 -emit-llvm %s -triple x86_64-unknown-linux-gnu -o - | FileCheck %s
```

Comment:
Consider using update_cc_test_checks.py.

I'd like to see checks that we emit convergence tokens on targets where that's the default.

Reply:
> Consider using update_cc_test_checks.py.
> 
> I'd like to see checks that we emit convergence tokens on targets where that's the default.

However, it seems that the script doesn't generate checks on attribute definitions. I tried '--check-attributes`, but it only added function attributes.

Reply:
I hacked `llvm/utils/UpdateTestChecks/common.py` a little bit to generate attribute def checks. I may ask for another review on that hack on update tools.

Reply:
Please don't merge an "auto-generated" file that isn't actually autogenerated by the in-tree version.

(Is --check-attributes not what you want?)

Reply:
> Please don't merge an "auto-generated" file that isn't actually autogenerated by the in-tree version.
> 
> (Is --check-attributes not what you want?)

yeah, `--check-attributes` only enable function attributes output per function. There's no still no attr definition output. I have to make the following change:

```
$ git diff
diff --git a/llvm/utils/UpdateTestChecks/common.py b/llvm/utils/UpdateTestChecks/common.py
index eb212ed304e9..2161a3589966 100644
--- a/llvm/utils/UpdateTestChecks/common.py
+++ b/llvm/utils/UpdateTestChecks/common.py
@@ -2156,10 +2156,6 @@ def filter_globals_according_to_preference(
         return global_val_lines_w_index
     assert global_check_setting == "smart"
 
-    if nameless_value.check_key == "#":
-        # attribute sets are usually better checked by --check-attributes
-        return []
-
     def extract(line, nv):
         p = (
             "^"
```
Shall I use the original test check? We need to check attribute definitions including 'convergent'.

Reply:
Update the test again with '--check-globals=all'. That output is the expected one.